extension/goext/environment.go

```diff
@@ -193,3 +233,35 @@ func MakeNullFloat(value float64) NullFloat {
 		Valid: true,
 	}
 }
+
+// MakeUndefinedString allocates a new undefined string and sets its value
+func MakeUndefinedString(value string) *NullString {
+	return &NullString{
+		Value: value,
+		Valid: true,
+	}
+}
+
+// MakeUndefinedBool allocates a new undefined bool and sets its value
+func MakeUndefinedBool(value bool) *NullBool {
```

Comment:
The approach for tri-state values using pointer is dangerous in my opinion. In golang, bool, int and string types are value based whereas the newly introduced generalizations are pointer based. It means that the semantics of copy operation changed fundamentally:
```
var a *NullBool = MakeUndefinedBool(false)
b := a   // <- intuition is that we copy value, but in fact we copy pointer
a.Value = true
fmt.Println("%b", b.Value)    // <- returns true
```

Reply:
I agree. Value based is better approach. Please see Maybe structs